clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -460,15 +463,21 @@ boolean joinGroupIfNeeded(final Timer timer) {
                     exception instanceof IllegalGenerationException ||
                     exception instanceof MemberIdRequiredException)
                     continue;
-                else if (!future.isRetriable())
-                    throw exception;
-
-                timer.sleep(rebalanceConfig.retryBackoffMs);
+                else {
+                    handleFailure(future, timer);
+                }
             }
         }
         return true;
     }
 
+    protected void handleFailure(RequestFuture<?> future, Timer timer) {
+        if (future.isRetriable() || future.exception() instanceof AuthorizationException)
```

Comment:
@hachikuji AuthorizationException is not marked as a retriable exception, but I think applications would typically retry since it could be transient while ACLs have not yet been propagated. Not sure whether we should do one or both of these:
1) Make AuthorizationException retriable since it could be transient, seemed riskier than just checking here.
2) Add backoff for all exceptions, not just AuthorizationException and other retriable exceptions to avoid tight loops. Wasn't sure what other fatal exceptions may be thrown when looking up coordinator.

Reply:
GroupAuthorizationException is the only retriable exception defined in the ticket, maybe we should just scope the fix only to that. Agree with No.2 to backoff for all retriable exceptions.

Reply:
@abbccdda Thanks for the review. The instance we saw was an authorization failure for groups, but since I was fixing handling of all responses, I used `AuthorizationException` in the check since we want to ensure we backoff for topic authorization exceptions too. We already backoff for `RetriableException` and I haven't changed that behaviour, this PR addresses exceptions that are not subclasses of `RetriableException`. Perhaps just updating `AuthorizationException` as is currently done in the PR is sufficient?

Reply:
Yes, I think the current PR scope is fine for just fixing consumer issue. We should just be careful not to accidentally make retries for other components relying on subclass of `AuthorizationException`. Could we reorder the check though, to make throw behavior happen first?

Reply:
`AuthorizationException` is not retriable and I kept that it that way, so this PR won't affect other components. Since AuthorizationException is not retriable, we want to sleep first before throwing the exception, hence the ordering of sleep before throw.

Reply:
@abbccdda Thinking about this a bit more, it seems safer to restrict backoff to GroupAuthorizationException as you had mentioned earlier in this thread. Consumers requests with topics may contain multiple topics of which only a subset are unauthorized. We probably don't want to apply backoff in that case. Since the issue we have seen is only with group authorization, I have updated the PR to only apply backoff for GroupAuthorizationException (in addition to retriable exceptions which we were handling earlier).